java-api-core/src/com/kinvey/java/store/UserStoreRequestManager.java

```diff
@@ -751,7 +752,9 @@ public LoginRequest(Credential credential) {
 
         public LoginRequest buildAuthRequest() {
             this.request = builder.build();
-            this.request.setKinveyHeaders(((KinveyClientRequestInitializer) client.getKinveyRequestInitializer()).getKinveyHeaders());
+            KinveyHeaders kinveyHeaders = (((KinveyClientRequestInitializer) client.getKinveyRequestInitializer()).getKinveyHeaders());
+            kinveyHeaders.set("X-Kinvey-Client-App-Version", clientAppVersion);
+            this.request.setKinveyHeaders(kinveyHeaders);
```

Comment:
We should only be setting the **X-Kinvey-Client-App-Version** header if the **clientAppVersion** has been set by the developer. If the **clientAppVersion** is null or empty, the header should not be added.

Reply:
Updated